src/scanner.cc

```diff
@@ -271,6 +275,14 @@ struct Scanner {
     }
   }
 
+  
+  // This function returns true if c is a valid starting character of a name/identifier
+  bool is_identifier_start_char(char c){
```

Comment:
I need to add an auto-formatter, so sorry about these nits: but can we change `){` to `) {` and remove the extra white space line on line 278 (there's two consecutive blank lines in 277 and 278).

Reply:
Good catch! For some reason I did not have the auto-format on save enabled in VSCode but it should be good after enabling it.